File: Support Vector Regression (SVR)/svr.R
# read dataset
df = read.csv('Position_Salaries.csv')

#treat missing data --> not needed here
#treat categorical data --> not needed here
df = df[2:3]

#split into training and testing data --> not needed since dataset size is small
#SVR
#install.packages('e1071')
library(e1071)
regressor = svm(formula = Salary~., data = df, type = 'eps-regression')

#predict new data
y_pred = predict(regressor, newdata = data.frame(Level = 6.5))

#plot values
library(ggplot2)
ggplot()+
  geom_point(aes(x = df$Level, y = df$Salary),colour='red')+
  geom_line(aes(x = df$Level, y = predict(regressor,newdata = df)),colour = 'blue')+
  ggtitle('SVR visualization')+
  xlab('Position')+
  ylab('Salary')

File: Support Vector Regression (SVR)/svr.py
# -*- coding: utf-8 -*-

#import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

#import data
df = pd.read_csv('Position_Salaries.csv')

#treat missing data --> not needed here
#treat categorical data --> not needed here

X = df.iloc[:,1:2].values
y = df.iloc[:,2].values

#feature scaling
from sklearn.preprocessing import StandardScaler
x_sc = StandardScaler()
y_sc = StandardScaler()
X = x_sc.fit_transform(X)
y = y_sc.fit_transform(y)

#svr implementation
from sklearn.svm import SVR
regressor = SVR(kernel = 'rbf') #using rbf(Gaussian kernel)
regressor.fit(X,y)


#predict output of new data
y_pred = y_sc.inverse_transform(regressor.predict(x_sc.transform(np.array([[6.5]]))))
#plot data
plt.scatter(X,y,color='red')
plt.plot(X,regressor.predict(X),color='blue')
plt.xlabel('Position')
plt.ylabel('Salary')
plt.title('SVR regression')
plt.show()
